### ISSUE TITLE
Robot misaligned with QR codes in the QR-code demo, detection fails

### ISSUE THREAD

**Issue by Minipada** [Minipada/ros2_data_collection#51]
The robot sometimes stops misaligned with a QR code and detection fails.

## Status after the Harmonic migration

Filed against Gazebo Classic. #268 ported the simulation to gz-sim, and #279 is porting this demo's launch path (including a known `robot_sdf` pass-through bug in `dc_demos/launch/tb3_qrcodes.launch.py`, where the argument is declared but never forwarded to `nav2_bringup`, so the demo may not even be spawning the intended robot model). Alignment must be re-assessed once that lands — a demo spawning the wrong robot description would produce exactly this symptom.

## Likely causes to check, in order

- `robot_sdf` / robot description actually used by the spawned robot (see #279)
- Nav2 goal tolerance (`xy_goal_tolerance` / `yaw_goal_tolerance`) vs. the camera's field of view at the QR-code standoff distance
- Waypoint poses in `qrcodes_waypoint_follower.py` vs. the actual prop placement in `qrcodes.world`
- AMCL convergence against `qrcodes.pgm` before the first waypoint

## Acceptance criteria

- [ ] Detection succeeds on every QR code across several consecutive demo runs
- [ ] Root cause documented (tolerance vs. world layout vs. description), not just tuned away

## Blocked by

- #279

**Comment by Minipada** [comment]
Still relevant — the QR-code inspection demo carries over to DC 2.0 and gets reworked in the demos slice (#251); fixing this belongs there or right after.

**Comment by Minipada** [comment]
Fixed by #336 (merged into \`jazzy\`), which found and corrected three independent misalignment sources: doubled camera-link/sensor pose offsets in \`waffle.model\` and \`dc_description\`'s URDF, non-unit waypoint quaternions and mis-placed waypoint rows, and an oversized \`yaw_goal_tolerance\`. Root cause, fix, and full verification (135-pose sweep, boundary sweep, live demo run, and a new \`detect\` regression stage in \`run.sh\`/CI) are documented in that PR's description and in \`progress.txt\`.

Closing manually since \`Closes #51\` only auto-closes on merge to this repo's default branch (\`humble\`), and #336 merged into \`jazzy\`.
